src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -1624,13 +1624,13 @@ class spell_dru_travel_form : public SpellScriptLoader
     static uint32 GetFormSpellId(Player const* player, Difficulty difficulty, bool requiresOutdoor)
     {
         // Check what form is appropriate
-        if (player->IsInWater()) // Aquatic form
+        if (player->HasSpell(SPELL_DRUID_FORM_AQUATIC) && player->IsInWater()) // Aquatic form
             return SPELL_DRUID_FORM_AQUATIC;
 
         if (!player->IsInCombat() && player->GetSkillValue(SKILL_RIDING) >= 225 && CheckLocationForForm(player, difficulty, requiresOutdoor, SPELL_DRUID_FORM_FLIGHT) == SPELL_CAST_OK) // Flight form
             return player->GetSkillValue(SKILL_RIDING) >= 300 ? SPELL_DRUID_FORM_SWIFT_FLIGHT : SPELL_DRUID_FORM_FLIGHT;
 
-        if (CheckLocationForForm(player, difficulty, requiresOutdoor, SPELL_DRUID_FORM_STAG) == SPELL_CAST_OK) // Stag form
+        if (!player->IsInWater() && CheckLocationForForm(player, difficulty, requiresOutdoor, SPELL_DRUID_FORM_STAG) == SPELL_CAST_OK) // Stag form
             return SPELL_DRUID_FORM_STAG;
 
         return 0;
```

Comment:
Shouldn't we actually also `PreventDefaultAction()` in `OnRemove` when we're returning 0 here or what is happening with 0 as triggerSpellID? (imagine a druid lvl 10 in water)  
Ref: https://www.youtube.com/watch?v=jMUE6cWcDvU

Reply:
fyi above posted bug only disappears when reapplying travelform after aquatic form is learned
Ref: https://www.youtube.com/watch?v=uY143NdOVWY

Reply:
> Shouldn't we actually also PreventDefaultAction() in OnRemove when we're returning 0 here or what is happening with 0 as triggerSpellID? (imagine a druid lvl 10 in water)

This is a static method, a "utility" one, and can't actually perform `PreventDefaultAction()` in it.
However it will be called if the current spell id is also 0 (hence no form).

 > If we'd want to be 100% blizzlike we'd also have to implement this bug (enter water, use travelform, run out of water and use mount):
Yes I remember that funny bit xd I'm assuming that bug will be fixed at some point and it doesn't really give anything, so might as well make it work on tc now.
That is my opinion though.